rust/agama-server/src/web/event.rs

```diff
@@ -8,13 +8,28 @@ use tokio::sync::broadcast::{Receiver, Sender};
 #[serde(tag = "type")]
 pub enum Event {
     L10nConfigChanged(LocaleConfig),
-    LocaleChanged { locale: String },
-    Progress(Progress),
-    ProductChanged { id: String },
+    LocaleChanged {
+        locale: String,
+    },
+    Progress {
+        service: String,
+        #[serde(flatten)]
+        progress: Progress,
+    },
+    ProductChanged {
+        id: String,
+    },
     PatternsChanged(HashMap<String, SelectedBy>),
-    InstallationPhaseChanged { phase: InstallationPhase },
-    BusyServicesChanged { services: Vec<String> },
-    ServiceStatusChanged { service: String, status: u32 },
+    InstallationPhaseChanged {
+        phase: InstallationPhase,
+    },
+    BusyServicesChanged {
+        services: Vec<String>,
+    },
+    ServiceStatusChanged {
+        service: String,
+        status: u32,
+    },
```

Comment:
I expect that they usually come together that Busy and ServiceStatus change?

Reply:
They are different things: one is the list of busy services, which is only kept by the `Manager`. The `ServiceStatusChanged` belongs to each separate service. However, I am thinking that perhaps we are not interested in the `BusyServicesChanged` event from the web UI. I will check.